### ISSUE TITLE
Validation file parameter in the training script

### ISSUE THREAD

**Issue by 4turkuaz** [yafuly/DeepfakeTextDetect#7]
Hi,

I've been trying to re-run your experimental setup for a specific setting. One thing making me confused is that although in `train.sh` there is a validation file parameter, it's populated with `test.csv` as shown below, not the `valid.csv`.

```sh
...

valid_file="$data_path/test.csv"

...
````

Once I change the parameter from `test.csv` to `valid.csv`, accuracy improved around +2%. Even though I have read the related parts of the code in `main.py`, I wasn't able to find an explanation why this parameter is set as test file, and why the validation file isn't used.

Any help would be very appreciated.

**Comment by yafuly** [comment]
Hello,

Thank you for highlighting this issue. It seems there was a mistake during the upload of the training script. For our experiments, we actually used "valid.csv" for validation, similar to your approach. Specifically, our training process involved the Longformer detector over 5 epochs, and we selected the checkpoint that demonstrated the best validation performance.

Regarding your mention of a "2%" performance improvement, could you clarify if this means your results are 2% higher than those reported in our paper? This difference might be attributed to the variability introduced by different seeds. In our upcoming paper version, which will be published post the anonymity period, we address the impact of randomness in data split seeds and run seeds.

We have included the training state of our Longformer detector for your reference. We hope this information assists in better aligning the performance metrics.
[trainer_state.json](https://github.com/yafuly/DeepfakeTextDetect/files/13576513/trainer_state.json)

**Comment by 4turkuaz** [comment]
TL;DR No, not higher than the result reported in the paper.

For domain and model specific setting, I chose ELI5 dataset to train the model.
* Case 1: `valid_file` param is `test.csv`
* Case 2: `valid_file` param is `valid.csv`

Case 2 eval accuracy = 0.9835 > Case 1 eval accuracy = 0.9679 ---- 1.56%

This is what I mean by the improvement.

**Comment by yafuly** [comment]
Let me clarify your question:

1) If the "eval accuracy" refers to validation performance:
I think it's natural to see a validation performance gap if you use two different datasets, especially when the dataset is small (i.e., ELI5 subset only).

2) if the "eval accuracy" refers to test performance of the best validation checkpoint:
That's interesting. One thing that comes to my mind is the evaluation metrics. Specifically, if you choose the checkpoint with the lowest validation loss, there is a chance that you miss the checkpoint with the best accuracy. Maybe you can try selecting the checkpoint with the best accuracy and compare the two settings accordingly.
